dbms/src/Columns/ColumnAggregateFunction.cpp

```diff
@@ -65,23 +84,57 @@ MutableColumnPtr ColumnAggregateFunction::convertToValues() const
         *   AggregateFunction(quantileTiming(0.5), UInt64)
         * into UInt16 - already finished result of `quantileTiming`.
         */
-    if (const AggregateFunctionState * function_state = typeid_cast<const AggregateFunctionState *>(func.get()))
+
+    /** Convertion function is used in convertToValues and predictValues
+        * in the similar part of both functions
+        */
+
+    MutableColumnPtr res;
+    if (convertion(&res))
     {
-        auto res = createView();
-        res->set(function_state->getNestedFunction());
-        res->data.assign(data.begin(), data.end());
         return res;
     }
 
-    MutableColumnPtr res = func->getReturnType()->createColumn();
-    res->reserve(data.size());
-
     for (auto val : data)
         func->insertResultInto(val, *res);
 
     return res;
 }
 
+MutableColumnPtr ColumnAggregateFunction::predictValues(Block & block, const ColumnNumbers & arguments) const
+{
+    MutableColumnPtr res;
+    if (convertion(&res))
+    {
+        return res;
+    }
+
+    auto ML_function_Linear = typeid_cast<AggregateFunctionMLMethod<LinearModelData, NameLinearRegression> *>(func.get());
```

Comment:
It would be better to avoid hard-coding the function list

Reply:
Agree. Function should know it's type.

Reply:
I will add `predictValues()` to base class `IAggregateFunction`